src/sentry/utils/sentry_apps/webhooks.py

```diff
@@ -32,6 +32,7 @@
 
 P = ParamSpec("P")
 R = TypeVar("R")
+T = TypeVar("T", bound=Mapping[str, Any])
```

Comment:
I _think_ this can be simplified a bit using the new Python 3.12 syntax for generics: https://github.com/getsentry/sentry/blob/4fbecea3d3b2b0ccca9b81390190c631516b2455/src/sentry/hybridcloud/outbox/category.py#L249

Though you might still need this `TypeVar` four the `Mapping` type bound.